sotodlib/site_pipeline/update_obsdb.py

```diff
@@ -0,0 +1,160 @@
+            continue
+        if os.path.exists(os.path.join(dirpath, "M_index.yaml")):
+            #Looks like a context file
+            bookcart.append(dirpath)
+    #Check the books for the observations we want
+
+    for bookpath in bookcart:
+        if check_meta_type(bookpath) in accept_type:
+            """ obsfiledb creation
+            The extra files give us a bit of trouble. We'll assume that index,
+            book, and bookbinder_log are always here, and edit our config file
+            to cover the rest as extra files.
+            """
+            book_file_list = os.listdir(bookpath)
+            book_file_list.remove("M_index.yaml")
+            book_file_list.remove("M_book.yaml")
+            book_file_list.remove("Z_bookbinder_log.txt")
+            book_file_list = [file_name for file_name in book_file_list if not file_name.endswith(".g3")]
+            if len(book_file_list)>0:
+                configs["extra_extra_files"] += book_file_list
+                with open(config, "w") as config_file:
+                    yaml.dump(configs, config_file)
+                checkbook(bookpath, config, add=True)
+                configs["extra_extra_files"] = ["Z_bookbinder_log.txt"]
+                with open(config, "w") as config_file:
+                    yaml.dump(configs, config_file)
+
+            else:
+                checkbook(bookpath, config, add=True)
+
+            index = yaml.safe_load(open(os.path.join(bookpath, "M_index.yaml"), "rb"))
+            obs_id = index.pop("book_id")
+            tags = index.pop("tags")
+            detsets = index.pop("detsets")
+
+            if "cols" in configs:
+                very_clean = {col:index[col] for col in iter(configs["cols"]) if col in index}
+            else:
+                col_list = []
+                clean = {key:val for key, val in index.items() if val is not None}
+                very_clean = {key:val for key, val in clean.items() if type(val) is not list}
+                for key, val in very_clean.items():
+                    col_list.append(key+" "+type(val).__name__)
+                bookcartobsdb.add_obs_columns(col_list)
+
+            if tags != ([] or [""]):
+                bookcartobsdb.update_obs(obs_id, very_clean, tags=tags)
+            else:
+                bookcartobsdb.update_obs(obs_id, very_clean)
+           
+        else:
+            bookcart.remove(bookpath)
+    if "obsdb" in configs:
+        bookcartobsdb.to_file(configs["obsdb"])
+    else:
+        bookcartobsdb.to_file("obsdb_from{}_to{}".format(tback, tnow))
+
+
+def get_parser(parser=None):
+    if parser is None:
+        parser = argparse.ArgumentParser()
+    parser.add_argument('--base_dir', type=str, 
+        help="base directory from which to look for books")
+    parser.add_argument("--config", help="ObsDb configuration file",
+        default="config.yaml", type=str)
+    parser.add_argument('--recency', default=2, type=float,
```

Comment:
I think we'll want to expand the "time range of searching" in the future but for this PR I'd just like to start with this one but make it optional. (have this be optional with default=None and have None be the default for the keyword argument above. If recency is None we then search the whole folder system.